#!/usr/bin/env python
PACKAGE = "velocity_smoother"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator() 
 
gen.add( "speed_lim_v",    double_t,  0,     "A double parameter",    1.5,       0,        20)
gen.add( "speed_lim_w",    double_t,  0,     "A double parameter",    4.5,       0,        20)
gen.add( "accel_lim_v",    double_t,  0,     "A double parameter",    1.0,       0,        20)
gen.add( "accel_lim_w",    double_t,  0,     "A double parameter",    4.5,       0,        20)
gen.add( "frequency",      double_t,  0,     "A double parameter",    100.0,     0,      1000)
gen.add( "decel_factor",   double_t,  0,     "A double parameter",    1.0,       0,         1)
gen.add( "Q_noise",        double_t,  0,     "A double parameter",    0.1,       0,        10)
gen.add( "R_noise",        double_t,  0,     "A double parameter",    0.5,       0,        10)
gen.add( "move_sample_N",  double_t,  0,     "A double parameter",    5.0,       0,       100)
gen.add( "odom_delay_s",   double_t,  0,     "A double parameter",   0.10,       0,        10)
gen.add( "dev_num_max",    double_t,  0,     "A double parameter",   20.0,       0,       100)
gen.add( "observe_time_s", double_t,  0,     "A double parameter",    2.0,       0,       100)
gen.add( "vel_dev",        double_t,  0,     "A double parameter",    0.4,       0,       100)
gen.add( "rot_dev",        double_t,  0,     "A double parameter",    0.4,       0,       100)
gen.add( "sum_x_dev",      double_t,  0,     "A double parameter",    0.5,       0,       100)
gen.add( "sum_y_dev",      double_t,  0,     "A double parameter",    0.5,       0,       100)
gen.add( "sum_w_dev",      double_t,  0,     "A double parameter",    0.5,       0,       100)

gen.add( "use_calmanFilter",   bool_t,    0,     "A Boolean parameter",    True)
gen.add( "use_movingFilter",   bool_t,    0,     "A Boolean parameter",    True)
gen.add( "use_limit_back",     bool_t,    0,     "A Boolean parameter",    True)
gen.add( "use_limit_speedup",  bool_t,    0,     "A Boolean parameter",   False)
gen.add( "use_fault_checked",  bool_t,    0,     "A Blloean parameter",    True)
gen.add( "is_debug",           bool_t,    0,     "A Boolean parameter",   False)

exit(gen.generate(PACKAGE, "velocity_smoother_node", "VelocitySmooth"))
